src/main/java/TaskList.java

```diff
@@ -0,0 +1,122 @@
+import java.util.ArrayList;
+
+/**
+ * TaskList class contains and facilitate operations on the inputTasks.
+ */
+public class TaskList {
+    private static ArrayList<Task> inputTasks = new ArrayList<>();
+    private static String acknowledgeMessage;
+
+    public static int getTaskListSize() {
+        return inputTasks.size();
+    }
+
+    public static ArrayList<Task> getInputTasks() {
+        return inputTasks;
+    }
+
+    /**
+     * Parses a given input String and stores it as a new task corresponding to todo/event/deadline.
+     *
+     * @param input              Command entered by the user.
+     * @param shouldAppendToFile Flag to indicate whether the parsed task should be appended to file.
+     * @throws DukeException If neither todo, deadline or event correspond to the input String.
+     */
+    public static void storeTask(String input, boolean shouldAppendToFile) throws DukeException {
+        Task newTask;
+        if (input.startsWith("todo")) {
+            newTask = new Todo(input.substring(4).trim());
+        } else if (input.startsWith("deadline")) {
+            String[] inputSlices = input.substring(8).trim().split("/");
+            newTask = new Deadline(inputSlices[0].trim(), inputSlices[1].substring(2).trim());
+        } else if (input.startsWith("event")) {
+            String[] inputSlices = input.substring(5).trim().split("/");
+            newTask = new Event(inputSlices[0].trim(), inputSlices[1].substring(2).trim());
+        } else {
+            throw new DukeException();
+        }
+
+        inputTasks.add(newTask);
+
+        acknowledgeMessage = "Got it. I've added this task: \n  " + inputTasks.get(inputTasks.size() - 1) + "\n"
+                + "Now you have: " + inputTasks.size() + " tasks in the list";
+
+        Ui.echo(acknowledgeMessage);
+
+        if (shouldAppendToFile) {
+            Storage.appendTaskToFile(newTask);
+        }
+    }
```

Comment:
See your comment: `Parses ... and stores ...`
So you can consider splitting the method to 2 different methods each handling its own functionality: parsing and storing.